java/code/src/com/suse/manager/webui/websocket/Notification.java

```diff
@@ -59,23 +61,25 @@
      */
     @OnOpen
     public void onOpen(Session session, EndpointConfig config) {
-        User user = (User)session.getUserProperties().get("currentUser");
         if (session != null) {
-            if (user != null) {
-                LOG.debug(String.format("Hooked a new websocket session [id:%s]", session.getId()));
-                handshakeSession(user, session);
-
-                // update the notification counter to the unread messages
-                try {
-                    sendMessage(session, String.valueOf(UserNotificationFactory.unreadUserNotificationsSize(user)));
-                }
-                finally {
-                    HibernateFactory.closeSession();
-                }
-            }
-            else {
-                LOG.debug("no authenticated user.");
-            }
+            Optional.ofNullable(session.getUserProperties().get("webUserID"))
+                    .map(webUserID -> UserFactory.lookupById((Long) webUserID))
+                    .ifPresentOrElse(user -> {
+                        // force to load user roles
+                        user.getRoles();
```

Comment:
Is this necessary now?

Reply:
you are correct, it's not necessary :) 
I refactor the code to not get the user from the database. We just need the user ID, which we already have from the session. Now we only go once to the database to get the number of notifications